use-shopping-cart/src/@types/use-shopping-cart/index.d.ts

```diff
@@ -0,0 +1,105 @@
+declare module 'use-shopping-cart' {
+
+    export interface ProviderProps {
+        children: JSX.Element,
+        stripe: any,
+        successUrl: string,
+        cancelUrl: string,
+        currency: string,
+        language: string,
+        billingAddressCollection: boolean,
+        allowedCountries: null | string[]
+    }
+
+    /**
+	 * Context provider to interact with Stripe API
+	 * @param children The Entry point to your app
+	 * @param stripe The stripe object
+	 * @param successUrl The redirect url for a successful sale
+	 * @param cancelUrl The redirect url for a cancelled sale
+	 * @param currency The preferred currency
+	 * @param language The language to be used in the checkout
+	 * @param billingAddressCollection Should the billing address be collected at the checkout. Defaults to false
+     * @param allowedCountries The allowed countries
+	 */
+    export const CartProvider: (props: ProviderProps) => JSX.Element;
+
+    export interface Product {
```

Comment:
A product doesn't have to have a name or an image, I believe all it actually needs is an `sku` and a `price`, however, the `Product` type should let you add more data to it in case you do want names and images maybe even alt attributes associated with each product.

Reply:
Yeah I saw that in the docs. I made name, image and currency optional with the (?).  

Ive now added `[propName: string]: any;` which should allow the user to add any other properties they wish as long as they are not any of the already declared properties. Of course by using any we are saying they could be anything. If you know what they could be i could create another interface to hold those explicitly and use that instead of any here.